tests/Moq.Tests/CustomTypeMatchersFixture.cs

```diff
@@ -229,5 +271,26 @@ public struct AnyStruct : ITypeMatcher
 		{
 			public bool Matches(Type typeArgument) => typeArgument.IsValueType;
 		}
+
+		public class ObjectStringOrdinalIgnoreCaseComparer : IEqualityComparer<object>
+		{
+			private static IEqualityComparer<string> InternalComparer => StringComparer.OrdinalIgnoreCase;
+
+			public new bool Equals(object x, object y)
+			{
+				var xType = x?.GetType();
+				var yType = y?.GetType();
+				Debug.Assert(xType == yType && xType == typeof(string));
```

Comment:
Small suggestion: Wouldn't it be simpler to do simply this:

```csharp
Debug.Assert(x is string && y is string);
```

(Since `string` cannot be subclassed, `x is string` does a precise type match and should be equivalent to `x?.GetType() == typeof(string)`.)

Reply:
Indeed. For the 2nd method I did that, it slipped my mind on this one. Thanks.

Reply:
@weitzhandler, FYI, I just noticed that this assertion actually fails during a test run, because a test (`It_Is_object_works_with_custom_comparer`) provides an argument of `(string)null`. I've pushed a commit removing this assertion altogether as it is too strict & not essential.

Reply:
Oh wow, I'm so sorry.
Is any any action from my side still required?

Reply:
No, I just pinged you because it was recently added code, and I wanted to give you a change to review, in case I missed something. But no worries, it's no big deal at all!

Reply:
Really appreciate your input, I'll look into it